packages/adapter-fauna/src/index.ts

```diff
@@ -10,140 +9,68 @@
-  Get,
-  If,
-  Index,
-  Let,
-  Match,
-  Ref,
-  Select,
-  Time,
-  Update,
-  Var,
-  Paginate,
-  Lambda,
-  Do,
-  Foreach,
-  errors,
-} from "faunadb"
+import { Client, TimeStub, fql } from "fauna"
 
-import {
+import type {
   Adapter,
-  AdapterSession,
   AdapterUser,
+  AdapterSession,
   VerificationToken,
+  AdapterAccount,
 } from "@auth/core/adapters"
 
-export const collections = {
-  Users: Collection("users"),
-  Accounts: Collection("accounts"),
-  Sessions: Collection("sessions"),
-  VerificationTokens: Collection("verification_tokens"),
-} as const
-
-export const indexes = {
-  AccountByProviderAndProviderAccountId: Index(
-    "account_by_provider_and_provider_account_id"
-  ),
-  UserByEmail: Index("user_by_email"),
-  SessionByToken: Index("session_by_session_token"),
-  VerificationTokenByIdentifierAndToken: Index(
-    "verification_token_by_identifier_and_token"
-  ),
-  SessionsByUser: Index("sessions_by_user_id"),
-  AccountsByUser: Index("accounts_by_user_id"),
-} as const
-
-export const format = {
-  /** Takes a plain old JavaScript object and turns it into a Fauna object */
-  to(object: Record<string, any>) {
-    const newObject: Record<string, unknown> = {}
-    for (const key in object) {
-      const value = object[key]
-      if (value instanceof Date) {
-        newObject[key] = Time(value.toISOString())
-      } else {
-        newObject[key] = value
-      }
-    }
-    return newObject
-  },
-  /** Takes a Fauna object and returns a plain old JavaScript object */
-  from<T = Record<string, unknown>>(object: Record<string, any>): T {
-    const newObject: Record<string, unknown> = {}
-    for (const key in object) {
-      const value = object[key]
-      if (value?.value && typeof value.value === "string") {
-        newObject[key] = new Date(value.value)
-      } else {
-        newObject[key] = value
-      }
-    }
-    return newObject as T
-  },
+type ToFauna<T> = {
+  [P in keyof T]: T[P] extends Date | null ? TimeStub | null : T[P]
 }
 
-/**
- * Fauna throws an error when something is not found in the db,
- * `next-auth` expects `null` to be returned
- */
-export function query(f: FaunaClient, format: (...args: any) => any) {
-  return async function <T>(expr: ExprArg): Promise<T | null> {
-    try {
-      const result = await f.query<{
-        data: T
-        ref: { id: string }
-      } | null>(expr)
-      if (!result) return null
-      return format({ ...result.data, id: result.ref.id })
-    } catch (error) {
-      if ((error as errors.FaunaError).name === "NotFound") return null
-      if (
-        (error as errors.FaunaError).description?.includes(
-          "Number or numeric String expected"
-        )
-      )
-        return null
+export type FaunaUser = ToFauna<AdapterUser>
+export type FaunaSession = ToFauna<AdapterSession>
+export type FaunaVerificationToken = ToFauna<VerificationToken>
+export type FaunaAccount = ToFauna<AdapterAccount> & any // TODO: Remove `& any`
```

Comment:
I could not drop the any here. Any ideas why?

Reply:
<img width="697" alt="Skærmbillede 2024-02-05 kl  10 37 15" src="https://github.com/nextauthjs/next-auth/assets/22524988/34fa3f89-12a1-469c-9710-afb6eacd2dd0">

It has to do with the last line in this interface (`readonly [parameter: string]: JsonValue | undefined;`). 
Even if I change it to be `readonly [parameter: string]: string;`, it still doesn't work (even though that should be compatible with Fauna's `QueryValue` type (`undefined` is not a valid type for `QueryValue`). If I remove the last line, however, the type error disappears.

Reply:
@employee451 could you take another look at this todo? Would be great to get that `any` out of there :pray:

Reply:
@ndom91 I have just pushed a fix for this. Please take a look